engines/osxnative/engine.go

```diff
@@ -12,22 +12,40 @@ import (
 
 type engine struct {
 	engines.EngineBase
-	log *logrus.Entry
+	config *configType
+	log    *logrus.Entry
 }
 
 type engineProvider struct {
 	engines.EngineProviderBase
 }
 
 func (e engineProvider) NewEngine(options engines.EngineOptions) (engines.Engine, error) {
-	return engine{log: options.Log}, nil
+	var c configType
+	err := configSchema.Map(options.Config, &c)
+	if err == schematypes.ErrTypeMismatch {
+		panic("Type mismatch")
+	} else if err != nil {
+		options.Log.WithError(err).Error("Invalid configuration")
+		return nil, engines.ErrContractViolation
+	}
```

Comment:
I haven't done this everywhere yet, but the pattern:
```
	var c configType
	if schematypes.MustMap(configSchema, options.Config, &c) != nil {
		return nil, engines.ErrContractViolation
	}
```
Is a bit simple... and panic is done is `MustMap`

See: https://github.com/taskcluster/taskcluster-worker/blob/master/plugins/interactive/interactive.go#L112-L115